Sources/SwiftKafka/Configuration/KafkaConfiguration+Security.swift

```diff
@@ -0,0 +1,453 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension KafkaConfiguration {
+    // MARK: - SSLConfiguration
+
+    /// Use to configure an SSL connection.
+    public struct SSLConfiguration: Sendable, Hashable {
+        /// A key either located in a file or directly as a key String (PEM format).
+        public struct Key: Sendable, Hashable {
```

Comment:
Probably the right term here would be `Chain` as it not only contains the public key but at least one certificate and possibly multiple certificates.

Reply:
Hm... maybe even two types.
- One of them is kind of a `LeafAndIntermediates` type as the first certificate in the chain will be used as the identity of the client and the remaining, in no particular order, will be used as potential intermediate certificates and send to the broker to verify the client.
- The other type could be `RootCertificates` as it is an unordered list of trusted CAs that is used by the client to verifier the identity of the broker.